src/core/middleware.js

```diff
@@ -20,11 +24,12 @@ export function prefixMiddleWare(req, res, next, { _config = config } = {}) {
 
   const hasValidLang = isValidLang(langFromURL);
   const hasValidClientApp = isValidClientApp(appFromURL, { _config });
+  const hasValidUrlException = isValidUrlException(appFromURL, { _config });
 
   let prependedLang = false;
 
   if ((hasValidLang && langFromURL !== lang) ||
-      (!hasValidLang && hasValidClientApp)) {
+      (!hasValidLang && (hasValidClientApp || hasValidUrlException))) {
```

Comment:
If I remove `|| hasValidUrlException` from the if statement, all tests still pass. Is it needed? Does it need test coverage?